File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Action/Common/RollMoveAction.ts
import { AbstractMonster } from "../../Unit/Monster/AbstractMonster";
import AbstractGameAction from "../AbstractGameAction";

export class RollMoveAction extends AbstractGameAction{
    public constructor(public enemy:AbstractMonster) {
        super();
    }

    async Update(): Promise<boolean> {
        if(this.enemy){
            this.enemy.RollMove();
        }
        return true;
    }
}

File: Unity/Assets/Scripts/ModTheSpire/Action/Combat/RollMoveAction.cs
using Cysharp.Threading.Tasks;
using ModTheSpire.Event;
using ModTheSpire.Unit;
using ModTheSpire.Unit.Monster;

namespace ModTheSpire.Effect.Other
{
    public class RollMoveAction : IGameAction
    {
        public float Duration { get; set; }
        public ActionType Type { get; } = ActionType.WAIT;
        private AbstractMonster mEnemy;
        public RollMoveAction(AbstractMonster enemy)
        {
            mEnemy = enemy;
        }
        public void Init()
        {
        }

        public UniTask<bool> Update()
        {
            if(mEnemy != null)
                mEnemy.RollMove();
            return UniTask.FromResult(true);
        }
    }
}